pip/resolve.py

```diff
@@ -173,6 +173,59 @@ def _check_skip_installed(self, req_to_install):
         self._set_req_to_reinstall(req_to_install)
         return None
 
+    def _get_abstract_dist_for(self, req):
```

Comment:
Done.

FWIW, I'll be adding type comments to various parts of the codebase once the mypy PR merges.